qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -0,0 +1,361 @@
+    is needed. At the minimum the developer must set the class variable :code:`__updater__` which
+    should have an :code:`update` method and can be chosen from the library
+    :mod:`qiskit_experiments.calibration_management.update_library`. See also
+    :class:`qiskit_experiments.calibration_management.update_library.BaseUpdater`. If no updater
+    is specified the experiment will still run but no update of the calibrations will be performed.
+
+    In addition to the calibration specific requirements, the developer must set the analysis method
+    with the class variable :code:`__analysis_class__` and any default experiment options.
+    """
+
+    # The updater class that updates the Calibrations instance. Different calibration
+    # experiments will use different updaters.
+    __updater__ = None
+
+    def __init_subclass__(cls, **kwargs):
+        """Warn if BaseCalibrationExperiment is not the first parent."""
+        for mro_cls in cls.mro():
+            if mro_cls is BaseCalibrationExperiment:
+                break
+            if issubclass(mro_cls, BaseExperiment) and not issubclass(
+                mro_cls, BaseCalibrationExperiment
+            ):
+                warnings.warn(
+                    "Calibration experiments must inherit from BaseCalibrationExperiment "
+                    f"before a BaseExperiment subclass: {cls}->{mro_cls}."
+                )
+                break
+        super().__init_subclass__(**kwargs)
+
+    # pylint: disable=super-init-not-called
+    def __init__(
+        self,
+        calibrations: Calibrations,
+        *args,
+        schedule_name: Optional[str] = None,
+        cal_parameter_name: Optional[str] = None,
+        auto_update: bool = True,
+        **kwargs,
+    ):
+        """Setup the calibration experiment object.
+
+        Args:
+            calibrations: The calibrations instance with which to initialize the experiment.
+            args: Arguments for the characterization class.
+            schedule_name: An optional string which specifies the name of the schedule in
+                the calibrations that will be updated.
+            cal_parameter_name: An optional string which specifies the name of the parameter in
+                the calibrations that will be updated. If None is given then no parameter will
+                be updated. Subclasses may assign default values in their init.
+            auto_update: If set to True (the default) then the calibrations will automatically be
+                updated once the experiment has run and :meth:`block_for_results()` will be called.
+            kwargs: Key word arguments for the characterization class.
+        """
+        super().__init__(*args, **kwargs)
+        self._cals = calibrations
+        self._sched_name = schedule_name
+        self._param_name = cal_parameter_name
+        self.auto_update = auto_update
+
+    def update_calibrations(self, experiment_data: ExperimentData):
```

Comment:
I'm wondering if this method should be part of the `Updater` itself and be called like `updater.update_calibrations(exp_data)`.  Since experiments can define custom updaters they could overload the updater if needed. I'm not sure if there would be issues with this though since the examples in this PR are pretty straightforward so nothing overloads this method.

If this was changed we should probably add `updater` as a property to return the  current updater too